src/device/all_gather.h

```diff
@@ -228,7 +228,7 @@ struct RunWorkColl<ncclFuncAllGather, T, RedOp, NCCL_ALGO_PAT, NCCL_PROTO_SIMPLE
 
     static constexpr int nworkers = NCCL_PAT_NWORKERS;
     struct ncclPatShmem* shmem = (struct ncclPatShmem*)ncclScratchForWarp(0);
-    uint64_t pollCount = 0;
+    // uint64_t pollCount = 0; unused variable - compiler warning
```

Comment:
An alternative would be this, which would remove the need for changes below:

```suggestion
    uint64_t pollCount = 0;
    (void)pollCount; // unused variable - compiler warning
```

Reply:
Any thoughts on which style we should prefer?